### PR TITLE
complete build for t-coffee

### LINKED ISSUE #38346: t-coffee package build is incomplete
In the process of [switching perl-bio-tools-run-alignment-tcoffee to use the t-coffee package](https://github.com/bioconda/bioconda-recipes/pull/38199) instead of the blacklisted t_coffee, we saw test failures with the execution of `t_coffee` (the executable):

I found the test command used in the perl library and tried running it directly with both packages:

# blacklisted t_coffee package:
```
$ t_coffee -profile cysprot1a.msf cysprot1b.msf -matrix blosum -method lalign_id_pair clustalw_pair -ktuple 3 -outfile testout -output clustalw -quiet

********************************************************************************
Command line arguments: ['/home/afif/miniconda3/envs/t_coffee/bin/t_coffee', '-profile', 'cysprot1a.msf', 'cysprot1b.msf', '-matrix', 'blosum', '-method', 'lalign_id_pair', 'clustalw_pair', '-ktuple', '3', '-outfile', 'testout', '-output', 'clustalw', '-quiet']
Install folder: /home/afif/miniconda3/envs/t_coffee/lib/t_coffee-11.0.8
********************************************************************************



********************************************************************************
Result: None
Error message: None
********************************************************************************
```

# current t-coffee package:
```
$ t_coffee -profile cysprot1a.msf cysprot1b.msf -matrix blosum -method lalign_id_pair clustalw_pair -ktuple 3 -outfile testout -output clustalw -quiet


*************************************************************************************************
*                        FULL TRACE BACK PID: 70061                                    
70076 -- ERROR: COREDUMP: T-COFFEE Version_13.45.0.4846264 (2020-10-15 17:52:11 - Revision 5becd5d - Build 620)
*************************************************************************************************

        ******************************************************************
        * Abnormal Termination
        * Job NOT Completed:[T-COFFEE, Version_13.45.0.4846264]
        * Please CHECK:                                       
        *       -1 The format of your Input Files                 
        *       -2 The parameters                                 
        *       -3 The use of special characters in sequence names:
        *                (@, |, %...)
        *       -4 The Online Doc (http://www.tcoffee.org)                   
        *       -5 re-run your CL (see below) with the -debug option. This will produce a debug file you can send us.
        *
        *           NO 
        * to:
        *               cedric.notredame@gmail.com
        * If you run T-Coffee over the WEB:
        *       Windows Cut and Paste is sometimes erratic and
        *       it can loose carriage returns. If you suspect this,
        *       try to cut and paste through an intermediate application
        *       (word pad) and inspect the results


        * CONFIDENTIALITY:
        *       The File NO may contain your personal DATA
        *       Remove ALL confidential DATA from this file BEFORE sending it
        ******************************************************************
# Command Line: /path/to/miniconda3/envs/tcoffee/share/t-coffee-13.45.0.4846264-4/bin/t_coffee -profile cysprot1a.msf cysprot1b.msf -matrix blosum -method lalign_id_pair clustalw_pair -ktuple 3 -outfile testout -output clustalw -quiet  [PROGRAM:T-COFFEE]

# TERMINATION STATUS: FAILURE [PROGRAM: T-COFFEE pid 70061 ppid 64028
#CL: /path/to/miniconda3/envs/tcoffee/share/t-coffee-13.45.0.4846264-4/bin/t_coffee -profile cysprot1a.msf cysprot1b.msf -matrix blosum -method lalign_id_pair clustalw_pair -ktuple 3 -outfile testout -output clustalw -quiet
```

### PR DESCRIPTION
**Pull request by 0xaf1f**
This build uses the developer-provided installer script to handle the complete build and installation.

closes #38346, which is a blocker for #38199, in turn needed to take care of #32576

### CODE REVIEW

**Comment by 0xaf1f** [comment]
Now the osx `prank` plugin is causing the build to fail...
```
20:54:35 BIOCONDA INFO (ERR) subprocess.CalledProcessError: Command '['/usr/local/miniconda/envs/bioconda/conda-bld/t-coffee_1670705170858/_build_env/bin/llvm-otool', '-l', '/usr/local/miniconda/envs/bioconda/conda-bld/t-coffee_1670705170858/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/libexec/t-coffee-13.45.0.4846264-5/plugins/macosx/prank']' returned non-zero exit status 1.
```

**Comment by 0xaf1f** [comment]
So in trying to debug the macosx, I'm stumped at this point. It's pretty difficult to debug not having a mac to work with.

The build.sh is executing the developer's build system, followed by some other commands that i want to postprocess. The developer's build system logs to stdout/stderr, and I did set -x on my build.sh so I could see when things are running.

It's looking like the developer's build system is somehow getting backgrounded part of the way through, because my post-processing commands start running (and failing because the build didn't finish yet and the files I expect to be there aren't there yet). The build system later completes successfully, but my postprocessing had already finished prematurely long before. then the build fails because of the lack of post-processing

[the build.sh script](https://github.com/bioconda/bioconda-recipes/blob/5d2332075d93bab4faa0f6eee540393e895b1258/recipes/t-coffee/build.sh)

[the whole build log](https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=25356&view=logs&jobId=1b052f1d-4456-52f0-9d43-71c4c5bd734d&j=1b052f1d-4456-52f0-9d43-71c4c5bd734d&t=edc48dcd-1fc2-5e3b-7036-7be9cea00123)

[developer build system starts](https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=25356&view=logs&j=1b052f1d-4456-52f0-9d43-71c4c5bd734d&t=edc48dcd-1fc2-5e3b-7036-7be9cea00123&l=1016)

[post-processing commands start](https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=25356&view=logs&j=1b052f1d-4456-52f0-9d43-71c4c5bd734d&t=edc48dcd-1fc2-5e3b-7036-7be9cea00123&l=2400) -- notice here that the build system is still not done. As a result, the post-processing fails because the files that should be there at this point don't exist yet.

[the error](https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=25356&view=logs&j=1b052f1d-4456-52f0-9d43-71c4c5bd734d&t=edc48dcd-1fc2-5e3b-7036-7be9cea00123&l=2814)

**Comment by 0xaf1f** [comment]
@BiocondaBot please add label

File: recipes/t-coffee/t_coffee.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (nsoranzo) [inline recipes/t-coffee/t_coffee.sh:1]:
```suggestion
#!/bin/sh
```

File: recipes/t-coffee/t_coffee.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
 set -e
 
 if [[ -f /proc/sys/kernel/pid_max ]] ; then
```
Comment (nsoranzo) [inline recipes/t-coffee/t_coffee.sh:4]:
```suggestion
if [ -f /proc/sys/kernel/pid_max ] ; then
```

**Comment by nsoranzo** [comment]
Since you've digged deep into this, maybe open an issue on https://github.com/cbcrg/tcoffee to track the build failures on macOS? Not necessary to merge this, just a bonus point :)

**Comment by 0xaf1f** [comment]
> Since you've digged deep into this, maybe open an issue on https://github.com/cbcrg/tcoffee to track the build failures on macOS?
I don't really understand what is going on on macosx. It's a failure that happens when conda build is wrapping up the package. It runs `llvm-otool -l` on all the binaries and that is what fails on two of the plugins. The most related thing I found was https://github.com/conda/conda-build/issues/4392

I added a commit with your suggested change. The build will take another ~2 hours and hopefully ready to merge by then.

**Review by nsoranzo — APPROVED** [review]
Thanks!

**Comment by 0xaf1f** [comment]
I don't think the automerge is working.

**Comment by nsoranzo** [comment]
@BiocondaBot please merge

**Comment by 0xaf1f** [comment]
I just realized that `t-coffee`'s dependency list is excessive and that regular usage doesn't require all those packages. It will cause similar issues as the old `t_coffee` did in terms of bloating the dependency chain. I'll make another PR...

**Pull request merged**